src/typelib.erl

```diff
@@ -104,6 +104,10 @@ remove_pos({type, _, 'fun', [{type, _, any}, RetTy]}) ->
     %% the only place where `{type, _, any}` can occure
     {type, erl_anno:new(0), 'fun', [{type, erl_anno:new(0), any}
                                    ,remove_pos(RetTy)]};
+remove_pos({type, _, 'fun', [{type, _, product, Args}, RetTy]}) ->
+    {type, erl_anno:new(0), 'fun', [
+        {type, erl_anno:new(0), product, [remove_pos(Arg) || Arg <- Args]}
+        , remove_pos(RetTy)]};
```

Comment:
The reason this wasn't here before is that `fun()` is normalized to `fun((...) -> any())`, which is equivallent. @gomoripeti did this some time ago. Maybe there are more places where `fun()` is not handled because it's expected to be represented as `fun((...) -> any())`?

I don't know, but if we only create new types on the form returned by normalize/2, we wouldn't need to add new clauses such as this.

Reply:
Sorry, my comment was wrong, so I made it hidden. I didn't read the code carefully.

I think this case should be covered by the function clause below: `remove_pos({type, _, Type, Params})`. Type would match 'fun' and then 'product' when called recursively. Doesn't it?

Reply:
Hum, I commented it and everything passes. I'm wondering if this is related to the fact I can't seem to get different arity functions with `is_function/2` tests to fail properly. I know I added this code during that time.

Reply:
I will remove it since looking at it, I'm pretty sure you're right.